### PR TITLE
Change LCD driver for clear license info

### PR DESCRIPTION
**Pull request by mangdangroboticsclub**
## Proposed change(s)

Change LCD driver for clear license info

### Useful links (GitHub issues, forum threads, etc.)

https://github.com/pimoroni/st7789-python/tree/master/library/ST7789

### Types of change(s)

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ *] Code refactor (non-breaking change which refactor the code)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update only
- [ ] Other (please describe)

## Testing and Verification

1. After re-install on Mini Pupper and Mini Pupper 2 based on Ubuntu22.04&Stanford, IP address can show
2. The LCD can display the pictures during playing

## Test Configuration

Mini Pupper, Mini Pupper 2

__Raspberry Pi OS version__  
Ubuntu 22.04 & StanfordQuadruped repo

## Checklist

- [ *] I have read the [CONTRIBUTING](https://github.com/mangdangroboticsclub/mini_pupper_2_bsp/blob/main/CONTRIBUTING.md) guidelines.
- [ *] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same change.

### CODE REVIEW

File: Python_Module/MangDang/LCD/ST7789.py [hunk 1]
```diff
@@ -1,180 +1,156 @@
...
+
 
 class ST7789(object):
-    """Representation of an ST7789 IPS LCD."""
+    """Representation of an ST7789 TFT LCD."""
+
+    def __init__(self, port, cs, dc, backlight=None, rst=None, width=240,
+                 height=240, rotation=90, invert=True, spi_speed_hz=4000000,
+                 offset_left=0,
+                 offset_top=0):
```
Comment (Tiryoh) [inline Python_Module/MangDang/LCD/ST7789.py]:
```suggestion
    def __init__(self, port=0, cs=0, dc=24, backlight=17, rst=27, width=320,
                 height=240, rotation=180, invert=True, spi_speed_hz=60000000,
                 offset_left=0, offset_top=0):
```

File: Python_Module/MangDang/mini_pupper/display.py [hunk 2]
```diff
@@ -8,9 +8,20 @@
...
+        dc=24, 
+        backlight=17,
+        rst=27,
+        width=320,
+        height=240,
+        rotation=180,
+        spi_speed_hz=60 * 1000 * 1000,
+        offset_left=0,
+        offset_top=0
+   )
```
Comment (Tiryoh) [inline Python_Module/MangDang/mini_pupper/display.py]:
```suggestion
    disp = ST7789()
```

**Review by Tiryoh — COMMENTED** [review]
The initial value can be set to maintain backward compatibility.
Is `disp.clear()` unnecessary?

File: Python_Module/MangDang/mini_pupper/display.py (hunk 2 above)
Comment (mangdangroboticsclub) [reply Python_Module/MangDang/mini_pupper/display.py]: Thanks for your comment.
1. I tested w/ and w/o disp.clear() , I have not found the difference.
2. If change ST7789.py but not display.py, do we need to add some comment info at the head of ST7789.py?

**Comment by mangdangroboticsclub** [comment]
About Python_Module/MangDang/mini_pupper/display.py    
    #with and without disp.clear(), have not found there is different on mini pupper, so delete it.
    #If comment it out, need to implement the display.clear() with several line code.
    #disp.clear()

**Comment by Tiryoh** [comment]
This seems to work with the current Mini Pupper hardware. Maybe someone needs to evaluate the performance in the future.

File: Python_Module/MangDang/LCD/ST7789.py [hunk 3]
```diff
@@ -1,180 +1,161 @@
...
-    # Keith (https://www.blogger.com/profile/02555547344016007163)
-    pb = np.array(image.convert('RGB')).astype('uint16')
-    color = ((pb[:,:,0] & 0xF8) << 8) | ((pb[:,:,1] & 0xFC) << 3) | (pb[:,:,2] >> 3)
-    return np.dstack(((color >> 8) & 0xFF, color & 0xFF)).flatten().tolist()
+
+import spidev
+import RPi.GPIO as GPIO
+
+
+__version__ = '0.0.4'
```
Comment (Tiryoh) [inline Python_Module/MangDang/LCD/ST7789.py]:
```suggestion
```

Is this version definition requried?

Comment (mangdangroboticsclub) [reply Python_Module/MangDang/LCD/ST7789.py]: no,I didn't find it is used, so delete it, thanks

**Pull request merged**